Site/SiteAnalyticsModule.php

```diff
@@ -146,11 +175,98 @@ public function hasAnalytics()
 		return (
 			$this->hasGoogleAnalytics() ||
 			$this->hasFacebookPixel() ||
+			$this->hasTwitterPixel() ||
 			$this->hasBingUET()
 		);
 	}
 
 	// }}}
+	// {{{ public function displayAnalytics()
+
+	public function displayAnalytics()
+	{
+		$this->displayGoogleAnalytics();
+		$this->displayFacebookPixelImage();
+		$this->displayTwitterPixelImages();
+		$this->displayBingUETImage();
+	}
+
+	// }}}
+	// {{{ public function getInlineJavascript()
+
+	public function getInlineJavascript()
+	{
+		$js = '';
+
+		if ($this->hasFacebookPixel()) {
+			$js.= $this->getFacebookPixelInlineJavascript();
+		}
+
+		if ($this->hasBingUET()) {
+			$js.= $this->getBingUETInlineJavascript();
+		}
+
+		if ($this->hasGoogleAnalytics()) {
+			$js.= $this->getGoogleAnalyticsInlineJavascript();
+		}
+
+		if ($this->hasTwitterPixel()) {
+			$js.= $this->getTwitterPixelInlineJavascript();
+		}
+
+		return $js;
+	}
+
+	// }}}
+	// {{{ protected function initOptOut()
+
+	protected function initOptOut()
+	{
+		$cookie_module = null;
+
+		if ($this->app->hasModule('SiteCookieModule')) {
+			$cookie_module = $this->app->getModule('SiteCookieModule');
+
+			if (isset($cookie_module->AnalyticsOptOut)) {
```

Comment:
This is old old old code that's been in the analytics module a long time. I just moved the method around.